src/plugin/file/kml/kmlparser.js

```diff
@@ -1195,38 +1195,46 @@ plugin.file.kml.KMLParser.prototype.readGroundOverlay_ = function(el) {
   goog.asserts.assert(el.localName == 'GroundOverlay', 'localName should be GroundOverlay');
 
   var obj = ol.xml.pushParseAndPop({}, plugin.file.kml.GROUND_OVERLAY_PARSERS, el, []);
-  if (obj && obj['extent'] && obj['Icon'] && obj['Icon']['href']) {
-    var icon = /** @type {string} */ (obj['Icon']['href']);
-    if (this.assetMap_[icon]) {
-      icon = this.assetMap_[icon]; // handle images included in a kmz
-    }
+  if (obj) {
+    if (obj['extent'] && obj['Icon'] && obj['Icon']['href']) {
+      var icon = /** @type {string} */ (obj['Icon']['href']);
+      if (this.assetMap_[icon]) {
+        icon = this.assetMap_[icon]; // handle images included in a kmz
+      }
 
-    var extent = ol.proj.transformExtent(/** @type {ol.Extent} */ (obj['extent']), os.proj.EPSG4326, os.map.PROJECTION);
+      var extent = ol.proj.transformExtent(/** @type {ol.Extent} */ (obj['extent']), os.proj.EPSG4326,
+          os.map.PROJECTION);
 
-    var feature = new ol.Feature();
-    this.setFeatureId_(feature);
+      var feature = new ol.Feature();
+      this.setFeatureId_(feature);
 
-    if (obj[os.data.RecordField.TIME]) {
-      feature.set(os.data.RecordField.TIME, obj[os.data.RecordField.TIME], true);
-    }
+      if (obj[os.data.RecordField.TIME]) {
+        feature.set(os.data.RecordField.TIME, obj[os.data.RecordField.TIME], true);
+      }
 
-    var image = new os.layer.Image({
-      source: new ol.source.ImageStatic({
-        url: icon,
-        imageExtent: extent
-      }),
-      url: icon
-    });
-    image.setId(/** @type {string} */ (feature.getId()));
+      var image = new os.layer.Image({
+        source: new os.source.ImageStatic({
+          crossOrigin: os.net.getCrossOrigin(icon),
+          url: icon,
+          imageExtent: extent,
+          projection: os.map.PROJECTION
+        }, -(obj['rotation'] || 0))
+      });
+      image.setId(/** @type {string} */ (feature.getId()));
 
-    var node = new plugin.file.kml.ui.KMLNode();
-    node.setFeature(feature);
-    node.setImage(image);
+      var node = new plugin.file.kml.ui.KMLNode();
+      node.setFeature(feature);
+      node.setImage(image);
 
-    return node;
-  } else {
-    return null;
+      return node;
+    } else if (obj['error']) {
+      var msg = obj['error'].toString();
+      goog.log.warning(plugin.file.kml.KMLParser.LOGGER_, msg);
+      os.alertManager.sendAlert(msg, os.alert.AlertEventSeverity.WARNING);
```

Comment:
This alert is in a function called to parse each `<GroundOverlay>` element, and could spam the user with repeated warnings if there are several non-rectangular LatLonQuad's present. This is probably an edge case, but it would be better to collect all the instances we can't handle and then fire one alert at the end.

Reply:
That logic is in the `AlertManager` itself. It defaults to no more than 5 duplicate messages, or we can call `sendAlertOnce` here instead for a single one.

Reply:
That's fair. I didn't think about the fact that the errors were all exact dupes. I still don't like depending on the AlertManager's dedupe logic in cases where we can control it because it's still super ugly when it happens. That should be mostly a fallback for unanticipated cases.